### DISCUSSION TITLE
Does dpdata not support custom data type?

### DISCUSSION THREAD

**Discussion by link89** [deepmodeling/dpdata#582] [General]
Recently I have a case that need to create `fparam.npy`. So I follow the example in this test case https://github.com/deepmodeling/dpdata/blob/master/tests/test_custom_data_type.py, but it turns out nothing is created.

**Comment by njzjz** [comment]
You also need to follow `tests/plugin`.

**Reply by link89** [reply]
I follow the code in `tests/plugin` and it doesn't work
```python
register_data_type(
    DataType("fparam", np.ndarray, (Axis.NFRAMES, -1), required=False), labeled=True,
)
```
However,  it works correctly with the following code

```python
dpdata.LabeledSystem.register_data_type(
    DataType("fparam", np.ndarray, (Axis.NFRAMES, -1), required=False)
)
```
My version is `dpdata==0.2.17`.

**Reply by njzjz** [reply]
`register_data_type` must be called before importing `LabeledSystem`. `dpdata.LabeledSystem.register_data_type` is used when `LabeledSystem` has been imported.

**Reply by link89** [reply]
I see, thank you. IMO it would be hard for a large project to guarantee the import order, perhaps `dpdata.LabeledSystem.register_data_type` should be the recommended way.
